commands/scale_cluster.go

```diff
@@ -38,108 +38,277 @@ Examples:
+func defaultScaleClusterArguments(cmd *cobra.Command, clusterId string, maxBefore int64, minBefore int64) scaleClusterArguments {
 	endpoint := config.Config.ChooseEndpoint(cmdAPIEndpoint)
 	token := config.Config.ChooseToken(endpoint, cmdToken)
 	scheme := config.Config.ChooseScheme(endpoint, cmdToken)
 
-	return scaleClusterArguments{
-		apiEndpoint:         endpoint,
-		authToken:           token,
-		scheme:              scheme,
-		clusterID:           cmdClusterID,
-		numWorkersDesired:   cmdNumWorkers,
-		workerNumCPUs:       cmdWorkerNumCPUs,
-		workerMemorySizeGB:  cmdWorkerMemorySizeGB,
-		workerStorageSizeGB: cmdWorkerStorageSizeGB,
-		oppressConfirmation: cmdForce,
-		verbose:             cmdVerbose,
+	scaleArgs := scaleClusterArguments{
+		apiEndpoint:           endpoint,
+		authToken:             token,
+		clusterID:             clusterId,
+		numWorkersDesired:     cmdNumWorkers,
+		oppressConfirmation:   cmdForce,
+		scheme:                scheme,
+		verbose:               cmdVerbose,
+		workersMax:            cmdWorkersMax,
+		workersMin:            cmdWorkersMin,
 	}
-}
 
-type scaleClusterResults struct {
-	// number of workers according to our info, just before the PATCH call.
-	numWorkersBefore int
-	// number of workers to add, just before the PATCH call. Might be negative.
-	numWorkersToAdd int
-	// number of workers as of the PATCH call response.
-	numWorkersAfter int
+	if !cmd.Flags().Changed(cmdWorkersMinName) {
+		scaleArgs.workersMin = minBefore
+	}
+	if !cmd.Flags().Changed(cmdWorkersMaxName) {
+		scaleArgs.workersMax = maxBefore
+	}
+	if !cmd.Flags().Changed(cmdWorkersMaxName) && !cmd.Flags().Changed(cmdWorkersMinName) && cmd.Flags().Changed("num-workers") {
+		scaleArgs.workersMax = int64(scaleArgs.numWorkersDesired)
+		scaleArgs.workersMin = int64(scaleArgs.numWorkersDesired)
+	}
+
+	return scaleArgs
 }
 
-func init() {
-	ScaleClusterCommand.Flags().BoolVarP(&cmdForce, "force", "", false, "If set, no confirmation is required when reducing the number of workers.")
-	ScaleClusterCommand.Flags().IntVarP(&cmdNumWorkers, "num-workers", "w", 0, "Number of worker nodes to have after scaling.")
-	ScaleClusterCommand.Flags().IntVarP(&cmdWorkerNumCPUs, "num-cpus", "", 0, "Number of CPU cores per added worker node.")
-	ScaleClusterCommand.Flags().Float32VarP(&cmdWorkerMemorySizeGB, "memory-gb", "", 0, "RAM per added worker node.")
-	ScaleClusterCommand.Flags().Float32VarP(&cmdWorkerStorageSizeGB, "storage-gb", "", 0, "Local storage size per added worker node.")
+// getClusterStatus returns the status for one cluster.
+func getClusterStatus(clusterID, activityName string) (*v1alpha1.StatusCluster, error) {
+	// perform API call
+	auxParams := ClientV2.DefaultAuxiliaryParams()
+	auxParams.ActivityName = activityName
 
-	ScaleCommand.AddCommand(ScaleClusterCommand)
-}
+	response, err := ClientV2.GetClusterStatus(clusterID, auxParams)
+	if err != nil {
+		if clientErr, ok := err.(*clienterror.APIError); ok {
```

Comment:
I would prefer having the usual error matchers as defined in https://github.com/giantswarm/fmt/blob/master/go/errors.md.

Reply:
I used the fmt way error matching now.